protocol/pubsub/v2/internal/connection_test.go

```diff
@@ -161,6 +161,60 @@ func TestPublishCreateTopic(t *testing.T) {
 	verifyTopicDeleteWorks(t, client, psconn, topicID)
 }
 
+// Test that publishing to an already created topic works and doesn't allow topic deletion
+func TestPublishWithCustomPublishSettings(t *testing.T) {
+	t.Run("create topic and publish to it with custom settings", func(t *testing.T) {
+		ctx := context.Background()
+		pc := &testPubsubClient{}
+		defer pc.Close()
+
+		projectID, topicID, subID := "test-project", "test-topic", "test-sub"
+
+		client, err := pc.New(ctx, projectID, nil)
+		if err != nil {
+			t.Fatalf("failed to create pubsub client: %v", err)
+		}
+		defer client.Close()
+
+		psconn := &Connection{
+			AllowCreateSubscription: true,
+			AllowCreateTopic:        true,
+			Client:                  client,
+			ProjectID:               projectID,
+			TopicID:                 topicID,
+			SubscriptionID:          subID,
+			PublishSettings: &pubsub.PublishSettings{
```

Comment:
How is topic deletion controlled by this setting?

Reply:
Hi @duglin my apologies, your confusion is a result of me copying and modifying one of the existing tests. I have corrected the test now.